helix-loader/src/lib.rs

```diff
@@ -3,14 +3,35 @@ pub mod grammar;
 
 use etcetera::base_strategy::{choose_base_strategy, BaseStrategy};
 use std::path::{Path, PathBuf};
+use std::sync::RwLock;
 
 pub const VERSION_AND_GIT_HASH: &str = env!("VERSION_AND_GIT_HASH");
 
+static CWD: RwLock<Option<PathBuf>> = RwLock::new(None);
+
 static RUNTIME_DIRS: once_cell::sync::Lazy<Vec<PathBuf>> =
     once_cell::sync::Lazy::new(prioritize_runtime_dirs);
 
 static CONFIG_FILE: once_cell::sync::OnceCell<PathBuf> = once_cell::sync::OnceCell::new();
 
+pub fn current_working_dir() -> PathBuf {
+    if let Some(path) = &*CWD.read().unwrap() {
+        return path.clone();
+    }
+
+    let cwd = std::env::current_dir().expect("Couldn't determine current working directory");
+    set_current_working_dir(cwd.clone()).expect("Couldn't set current working directory const");
```

Comment:
Not instead of calling the setter functionsi.only accessaccess the global directly here since that avoids an unnecessary call to `std::env::set_current_dir`